FireboltNETSDK/FireboltClient.cs

```diff
@@ -311,21 +322,65 @@ private async Task<T> SendAsync<T>(HttpMethod method, string uri, HttpContent? c
         }
         finally
         {
-            response.Dispose();
+            if (response != null)
+            {
+                response.Dispose();
+            }
         }
     }
 
 
 
-    private void AddAccessToken(HttpRequestMessage request)
+    private async void AddAccessToken(HttpRequestMessage request)
     {
         if (string.IsNullOrEmpty(_token))
         {
-            throw new FireboltException("The Access token is null or empty - EstablishConnection must be called");
+            _token = await EstablishConnection();
         }
         request.Headers.Add("Authorization", "Bearer " + _token);
     }
 
+    private static string ConvertToString(object? value, CultureInfo cultureInfo)
```

Comment:
Is this method used anywhere? I can't find references to it.

Reply:
You are right, this is not in use. This function was added in April

```
alexradzin@LenovoT14Firebolt:~/proj/firebolt-db/firebolt-net-sdk$ git blame ./FireboltNETSDK/FireboltClient.cs | grep ConvertToString
a4d7da2b (Aymeric           2023-04-03 17:15:27 +0100 343)     private static string ConvertToString(object? value, CultureInfo cultureInfo)
a4d7da2b (Aymeric           2023-04-03 17:15:27 +0100 373)                         return string.Join(",", array.Select(o => ConvertToString(o, cultureInfo)));

```